src/client/auth/mod.rs

```diff
@@ -262,12 +299,16 @@ impl Credential {
         };
 
         // Authenticate according to the chosen mechanism.
-        mechanism.authenticate_stream(conn, self).await
+        mechanism.authenticate_stream(conn, self, http_client).await
     }
 }
 
-pub(crate) fn generate_nonce() -> String {
+pub(crate) fn generate_nonce_bytes() -> Vec<u8> {
```

Comment:
We need the nonce as bytes in AWS auth, so I pulled this from `generate_nonce` into a separate method.